governator-legacy/src/main/java/com/netflix/governator/guice/InternalAutoBindModule.java

```diff
@@ -184,8 +186,14 @@ else if ( Module.class.isAssignableFrom(clazz) )
 
     private void bindAutoBindSingleton(AutoBindSingleton annotation, Class<?> clazz)
     {
-        LOG.info("Installing @AutoBindSingleton " + clazz.getName());
+        LOG.info("Installing @AutoBindSingleton '{}'", clazz.getName());
+        LOG.info("***** @AutoBindSingleton for '{}' is deprecated soon.\nPlease use a Guice module with bind({}.class).asEagerSingleton(); instead", 
+                clazz.getName(), clazz.getSimpleName() );
         
+        Singleton singletonAnnotation = clazz.getAnnotation(Singleton.class);
+        if (singletonAnnotation == null) {
+            LOG.info("***** {} should also be annotation with @Singleton to ensure singleton behavior", clazz.getName());
```

Comment:
Converting to property singleton's is tricky and a horrible hack.  We should go through a proper deprecation cycle here.  Also, we shouldn't fail fast since this will break existing code.